### PR TITLE
JIT: Delete some unused fields in `CallArgABIInformation`

### PR DESCRIPTION
**Pull request by jakobbotsch**

### CODE REVIEW

File: src/coreclr/jit/gentree.h [hunk 1]
```diff
@@ -4474,8 +4470,6 @@ struct CallArgABIInformation
     // Unix amd64 will split floating point types and integer types in structs
     // between floating point and general purpose registers. Keep track of that
     // information so we do not need to recompute it later.
```
Comment (filipnavara) [inline src/coreclr/jit/gentree.h:4472]: Maybe the comment should be updated too?

Comment (jakobbotsch) [reply src/coreclr/jit/gentree.h:4472]: I left it since the comment equally applies to the `StructDesc` that we do use in a few places.

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib, simple change

[No diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=609614&view=ms.vss-build-web.run-extensions-tab). superpmi-replay failure is a timeout.

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks!

**Pull request merged**